salt/metalk8s/kubeadm/init/addons/coredns.sls

```diff
@@ -0,0 +1,126 @@
+{%- from "metalk8s/map.jinja" import coredns with context %}
+
+{% set kubeconfig = "/etc/kubernetes/admin.conf" %}
+{% set context = "kubernetes-admin@kubernetes" %}
+
+Create coredns ConfigMap:
+  kubernetes.configmap_present:
+    - name: coredns
+    - namespace: kube-system
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }} 
+    - data:
+        Corefile: |
+          .:53 {
+              errors
+              health
+              kubernetes {{ coredns.cluster_domain }} {{ coredns.reverse_cidrs }} {
+                pods insecure
+                upstream
+                fallthrough in-addr.arpa ip6.arpa
+              }
+              prometheus :9153
+              proxy . /etc/resolv.conf
+              cache 30
+              loop
+              reload
+              loadbalance
+          }
+  require:
+    - pkg: Install Python Kubernetes client
+
+Create coredns deployment:
+  kubernetes.deployment_present:
+    - name: coredns
+    - namespace: kube-system
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - source: salt://metalk8s/kubeadm/init/addons/files/coredns_deployment.yaml
+  require:
+    - kubernetes: Create coredns ConfigMap
+    - pkg: Install Python Kubernetes client
+
+Create coredns service:
+  kubernetes.service_present:
+    - name: coredns
+    - namespace: kube-system
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - metadata:
+        name: kube-dns
+        namespace: kube-system
```

Comment:
You should remove those 2 metadata taken from the upper section and seems like you use `kube-dns` as name